cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -299,20 +299,15 @@ extern "C" int embed_sim_init(int argc, char const * const * argv)
         if (argv_item == NULL) {
             PyErr_Print();
             LOG_ERROR("Unable to convert command line argument %d to Unicode string.", i);
-            Py_DECREF(argv_list);
             goto cleanup;
         }
         PyList_SET_ITEM(argv_list, i, argv_item);                       // Note: This function steals the reference to argv_item
     }
 
-
     cocotb_retval = PyObject_CallFunctionObjArgs(cocotb_init, argv_list, NULL);
-    Py_DECREF(argv_list);
-    Py_DECREF(cocotb_init);
```

Comment:
I'm not really a fan of this change (read: this commit as a whole) - I think it's more instructive to see things cleaned up as soon as they're no longer needed

Reply:
That's a great sentiment, but that approach is more verbose, less likely to be correct, and more fragile, considering how the function is implemented. If we had a considerable amount of wrapper code, I might say we should use `std::unique_ptr` and introduce new scopes for scope-guard-style code.

Reply:
Would you mind popping that commit off so I can merge this, then I'll look at it again in a standalone PR?

Reply:
Thanks!